File: java_test/src/DesignPattern/Command/SimpleRemote/RemoteControlTest.java
package DesignPattern.Command.SimpleRemote;

/**
 * Created by Administrator on 2018/4/13.
 */
public class RemoteControlTest {
    public static void main(String[] args) {

        //构建打开灯命令
        Light light = new Light();
        LightOnCommand lightOn = new LightOnCommand(light);

        //构建车库门打开命令
        GarageDoor garageDoor = new GarageDoor();
        GarageDoorOpenCommand garageDoorOpen = new GarageDoorOpenCommand(garageDoor);

        /*-----------------------------------------------------------------------*/
        //简单遥控
        SimpleRemoteControl remote = new SimpleRemoteControl();

        //开灯
        remote.setCommand(lightOn);
        remote.buttonWasPressed();

        //开车库门
        remote.setCommand(garageDoorOpen);
        remote.buttonWasPressed();
    }
}

File: test_go/src/pattern/06_command/01_simple_romote/remote_control_test.go
package _1_simple_romote

/*
*  测试客户请求命令（请求调用者，和请求接收者之前解耦）
*/
import "testing"

func TestAll(t *testing.T) {

	//简单遥控
	var remote *SimpleRemoteControl = new(SimpleRemoteControl)

	//1.开灯
	var lightOn *LightOnCommand = NewLightOnCommand(new(Light)) //构建开灯命令
	remote.SetCommand(lightOn)
	remote.ButtonWasPressed()

	//2.开车库门
	var doorOpen *GarageDoorOpenCommand = NewGarageDoorOpenCommand(new(GarageDoor)) //构建打开车库门命令
	remote.SetCommand(doorOpen)
	remote.ButtonWasPressed()
}
